### PR TITLE
Feature/974 consolidate angular group templates

### PR DESCRIPTION
**Pull request by dheitzer**
consolidated templates into transaction-detail and double-transaction-detail components and added and replaced componentId with component group classes in models.

### CODE REVIEW

File: front-end/src/app/shared/models/transaction-groups/transaction-group-a.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      templateMap.aggregate,
+      templateMap.purpose_description,
+      templateMap.employer,
+      templateMap.occupation,
+      templateMap.memo_code,
+      templateMap.memo_text_input,
+      'subTransaction',
+    ].filter(val => !!val);
+  }
+
```
Comment (mjtravers) [inline front-end/src/app/shared/models/transaction-groups/transaction-group-a.ts]: The hope was that with the use of the templateMap we would not have to discern between the schedule A and schedule B when getting the form properties. For this case, schedule A, the only difference should be the category_code as the Group A has the employer and occupation by definition. Do we have Schedule B transaction types assigned to Group A that are individuals that do not collect the employer and occupation data?

File: front-end/src/app/shared/models/transaction-groups/transaction-group-a.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    return false;
+  }
+
+  hasElectionInformationInput(): boolean {
+    return false;
+  }
+
+  getAmountInputTitle(): string {
+    return 'Receipt Information';
+  }
```
Comment (mjtravers) [inline front-end/src/app/shared/models/transaction-groups/transaction-group-a.ts]: Is amountInputTitle something standard across schedule types and can be put in the templateMap?

File: front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.spec.ts [hunk 3]
```diff
@@ -304,7 +304,20 @@ describe('TransactionTypeBaseComponent', () => {
       life: 3000,
     };
     const messageServiceAddSpy = spyOn(testMessageService, 'add');
-    component.navigateTo(new NavigationEvent(NavigationAction.SAVE, NavigationDestination.ANOTHER));
+    spyOn(testRouter, 'navigateByUrl').and.callFake(
+      () =>
+        new Promise((res) => {
+          res(true);
+        })
+    );
```
Comment (mjtravers) [inline front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.spec.ts]: This promise is being reported as a code smell by SonarCloud. Let's look at SonarCloud's recommendation for changing the code.

File: front-end/src/app/shared/models/transaction-groups/transaction-group-a.spec.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      memo_code: '',
+      amount: '',
+      aggregate: '',
+      purpose_description: '',
+      purposeDescripLabel: '',
+      memo_text_input: '',
+      category_code: testCategoryCode,
+      election_code: '',
+      election_other_description: ''
+    }
```
Comment (mjtravers) [inline front-end/src/app/shared/models/transaction-groups/transaction-group-a.spec.ts]: There is a lot of code duplication defining these testTemplateMaps in each spec file and across all the transaction-group spec files. Can we migrate all of them into 1 or 2 definitions in the unit-test.utils.ts library?

File: front-end/src/app/shared/models/transaction-groups/transaction-group-a.ts (hunk 1 above)
Comment (dheitzer) [reply front-end/src/app/shared/models/transaction-groups/transaction-group-a.ts]: After discussing with you and Gregg yesterday, I went ahead and made this change to single templateMap.

File: front-end/src/app/shared/models/transaction-groups/transaction-group-a.ts (hunk 2 above)
Comment (dheitzer) [reply front-end/src/app/shared/models/transaction-groups/transaction-group-a.ts]: nice catch.  Went ahead and added this to templateMap.

File: front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.spec.ts (hunk 3 above)
Comment (dheitzer) [reply front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.spec.ts]: pulled in develop changes to fix this.

File: front-end/src/app/shared/models/transaction-groups/transaction-group-a.spec.ts (hunk 4 above)
Comment (dheitzer) [reply front-end/src/app/shared/models/transaction-groups/transaction-group-a.spec.ts]: used unit-test.utils as suggested and removed a lot of template def code.

**Review by mjtravers — APPROVED** [review]

**Pull request merged**